packages/kolibri-common/components/SyncSchedule/EditDeviceSyncSchedule.vue

```diff
@@ -0,0 +1,395 @@
+
+
+<script>
+
+  import ImmersivePage from 'kolibri.coreVue.components.ImmersivePage';
+  import BottomAppBar from 'kolibri.coreVue.components.BottomAppBar';
+  import { NetworkLocationResource, FacilityResource, TaskResource } from 'kolibri.resources';
+  import { now } from 'kolibri.utils.serverClock';
+  import commonCoreStrings from 'kolibri.coreVue.mixins.commonCoreStrings';
+  import { TaskTypes } from 'kolibri.utils.syncTaskUtils';
+  import { PageNames } from '../../../../kolibri/plugins/facility/assets/src/constants';
+
+  export default {
+    name: 'EditDeviceSyncSchedule',
+    components: {
+      ImmersivePage,
+      BottomAppBar,
+    },
+    mixins: [commonCoreStrings],
+    props: {
+      icon: {
+        type: String,
+        default: 'back',
+      },
+      msg: {
+        type: String,
+        default: '',
+      },
+    },
+    data() {
+      return {
+        removeDeviceModal: false,
+        deviceName: null,
+        device: [],
+        now: now(),
+        selectedItem: '',
+        tasks: [],
+        selectedDay: null,
+        selectedTime: null,
+      };
+    },
+    computed: {
+      backRoute() {
+        return { name: PageNames.ManageSyncSchedule };
+      },
+      pageHeight() {
+        return {
+          height: '80%',
+          zIndex: -1,
+        };
+      },
+      selectorStyle() {
+        return {
+          color: this.$themeTokens.text,
+          backgroundColor: this.$themePalette.grey.v_200,
+          borderRadius: '5px 5px 0px 0px',
+          paddingTop: '5px',
+          paddingLeft: '5px',
+          width: '300px',
+        };
+      },
+      selectArray() {
+        return [
+          { label: this.$tr('everyHour'), value: 3600 },
+          { label: this.$tr('everyDay'), value: 86400 },
+          { label: this.$tr('everyWeek'), value: 86400 },
+          { label: this.$tr('everyTwoWeeks'), value: 86400 },
+          { label: this.$tr('everyMonth'), value: 2592000 },
+        ];
+      },
+      getDays() {
+        const today = new Date();
+        const daysOfWeek = [];
+        const date = new Date(
+          today.getFullYear(),
+          today.getMonth(),
+          today.getDate() + (7 - today.getDay())
+        );
+        for (let i = 0; i < 7; i++) {
+          daysOfWeek.push({ label: this.$formatDate(date, { weekday: 'long' }), value: i });
+          date.setDate(date.getDate() + 1);
+        }
+        return daysOfWeek;
+      },
+
+      SyncTime() {
+        const endTime = new Date();
+        endTime.setHours(24, 0, 0, 0);
+        const interval = 30;
+
+        const times = [];
+        const time = new Date();
+        time.setHours(0, 0, 0, 0);
+
+        while (time < endTime) {
+          times.push(this.$formatTime(time));
+
+          time.setMinutes(time.getMinutes() + interval);
+        }
+
+        return times;
+      },
+    },
+    beforeMount() {
+      this.fetchDevice();
+    },
+    beforeDestroy() {
+      clearInterval(this.timer);
```

Comment:
I think this was copied in to clean up a timer to keep the `now` value refreshed - I think the logic came from the `ElapsedTime` component. As you are displaying the server time to the user here, it would be a good idea to keep it updating!

See the `mounted` lifecycle hook there: https://github.com/learningequality/kolibri/blob/develop/kolibri/core/assets/src/views/ElapsedTime.vue#L34 (you can probably copy it in verbatim).